src/main/java/com/ea/restaurant/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.exceptions.EntityNotFoundException;
+import com.ea.restaurant.exceptions.UserNotFoundException;
+import com.ea.restaurant.repository.UserRepository;
+import com.ea.restaurant.service.UserService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+  @Autowired private UserRepository userRepository;
+
+  @Override
+  public User save(User user) {
+    return userRepository.save(user);
+  }
+
+  @Override
+  public List<User> findAllEnabled() {
+    return userRepository.findAllByEntityStatus(Status.ACTIVE);
+  }
+
+  @Override
+  public User findEnabledById(Long id) {
+    return userRepository
+        .findByIdAndEntityStatus(id, Status.ACTIVE)
+        .orElseThrow(() -> new EntityNotFoundException(id));
```

Comment:
Shouldn't the developer that is using the service decide what to do if the user is not found?

Reply:
Done.